rust/lib.rs

```diff
@@ -178,6 +238,15 @@ impl VectorType for f32 {
     fn add(index: &Index, key: u64, vector: &[Self]) -> Result<(), cxx::Exception> {
         index.inner.add_f32(key, vector)
     }
+    fn add_raw(
+        _index: &Index,
+        _key: u64,
+        _vector: &[Self],
+        _element_bits: usize,
+    ) -> Result<(), cxx::Exception> {
+        ffi::throw_not_implemented_exception()?;
```

Comment:
does this function actually return? I'd think it would be `-> !` so the nest line would not be necessary